src/scene_battle_rpg2k3.cpp

```diff
@@ -2133,6 +2206,58 @@ Scene_Battle_Rpg2k3::BattleActionReturn Scene_Battle_Rpg2k3::ProcessBattleAction
 		}
 	}
 
+	if (action->GetCBAMovement() != lcf::rpg::BattlerAnimationItemSkill::Movement_none) {
+		SetBattleActionState(BattleActionState_MoveBack);
+	} else {
+		SetBattleActionState(BattleActionState_Execute);
+	}
+	return BattleActionReturn::eContinue;
+}
+
+Scene_Battle_Rpg2k3::BattleActionReturn Scene_Battle_Rpg2k3::ProcessBattleActionMoveBack(Game_BattleAlgorithm::AlgorithmBase* action) {
+	auto* source = action->GetSource();
+
+	if (source->GetType() == Game_Battler::Type_Ally) {
+		if (cba_move_frame == 0) {
+			auto* actor = static_cast<Game_Actor*>(source);
+			auto* sprite = actor->GetActorBattleSprite();
+			if (sprite) {
+				if (action->GetCBAMovement() == lcf::rpg::BattlerAnimationItemSkill::Movement_move) {
+					sprite->SetAnimationState(Sprite_Actor::AnimationState_WalkingRight);
+				} else {
+					sprite->SetAnimationState(Sprite_Actor::AnimationState_WalkingLeft);
+				}
+			}
+		}
+
+		if (cba_move_frame < 15) {
+			cba_move_frame++;
+			int frame = 15 - cba_move_frame;
+			int offset_x = 0;
+			int offset_y = 0;
+			if (action->GetCBAMovement() == lcf::rpg::BattlerAnimationItemSkill::Movement_step) {
+				offset_x = source->IsDirectionFlipped() ? frame * 2 : -frame * 2;
+			}
+			if (action->GetCBAMovement() == lcf::rpg::BattlerAnimationItemSkill::Movement_jump) {
+				offset_x = source->IsDirectionFlipped() ? frame * 2 : -frame * 2;
+				offset_y = -std::min(frame * 2, 30 - frame * 2);
+			}
+			if (action->GetCBAMovement() == lcf::rpg::BattlerAnimationItemSkill::Movement_move) {
+				offset_x = (cba_end_pos.x - cba_start_pos.x) * frame / 15;
+				offset_y = (cba_end_pos.y - cba_start_pos.y) * frame / 15;
+			}
+			source->SetBattlePosition(Point(cba_start_pos.x + offset_x, cba_start_pos.y + offset_y));
+			return BattleActionReturn::eWait;
+		} else {
+			auto* actor = static_cast<Game_Actor*>(source);
+			auto* sprite = actor->GetActorBattleSprite();
+			if (sprite) {
+				sprite->DoIdleAnimation();
+			}
+			cba_move_frame = 0;
+		}
+	}
```

Comment:
This whole block looks like copy and paste. Maybe we can make an `UpdateMovement()` function which both state handlers call and add some flag for the direction. Or you could add a multiplier, which is normally + or - 2.